sharktank/tests/utils/iree_test.py

```diff
@@ -115,3 +119,139 @@ def roundtrip(iree_devices: list[iree.runtime.HalDevice]):
             assert ops.equal(tensor, tensor_roundtrip)
 
         with_iree_device_context(roundtrip, iree_devices)
+
+
+COMPILE_FLAGS = [
+    "--iree-hal-target-device=local",
+    "--iree-hal-local-target-device-backends=llvm-cpu",
+]
+
+
+class SimpleModel(nn.Module):
+    def forward(self, x):
+        return torch.relu(x) + 1.0
+
+
+class MultiOutputModel(nn.Module):
+    def forward(self, x):
+        return torch.relu(x), torch.tanh(x)
+
+
+class TestCompileTorchModule:
+    """Tests for compile_torch_module_to_iree."""
+
+    def test_compilation(self, tmp_path):
+        """Test compilation with optional artifact saving."""
+        model = SimpleModel()
+        example_input = torch.randn(2, 32)
+        mlir_path = tmp_path / "model.mlir"
+        vmfb_path = tmp_path / "model.vmfb"
+
+        vmfb_bytes = compile_torch_module_to_iree(
+            model,
+            example_args=(example_input,),
+            compile_args=COMPILE_FLAGS,
+            save_mlir_to=mlir_path,
+            save_vmfb_to=vmfb_path,
+        )
+
+        assert len(vmfb_bytes) > 100
+
+        assert mlir_path.exists()
+        assert mlir_path.stat().st_size > 0
+
+        assert vmfb_path.exists()
+        assert vmfb_path.stat().st_size > 0
+
+
+class TestAdaptTorchModuleToIree:
+    """Tests for adapt_torch_module_to_iree."""
+
+    def test_basic_loading_and_execution(self):
+        """Test that loaded module executes and produces correct output shape."""
+        model = SimpleModel()
+        example_input = torch.randn(2, 32)
+
+        iree_module = adapt_torch_module_to_iree(
+            model,
+            example_args=(example_input,),
+            device="local-sync",
+            compile_args=COMPILE_FLAGS,
+        )
+
+        result = iree_module.forward(example_input)
+        assert isinstance(result, torch.Tensor)
+        assert result.shape == (2, 32)
+        assert not torch.isnan(result).any()
+
+    def test_output_matches_torch(self):
+        """Test that IREE output matches torch output."""
+        torch.manual_seed(42)
```

Comment:
You could use the [deterministic_random_seed](https://github.com/nod-ai/shark-ai/blob/b65fdae35ede615932734d6c204771a33cfdc7ec/sharktank/conftest.py#L449) pytest fixture. After the test it will return the RNG state to what was before the test.